staking.md

```diff
@@ -0,0 +1,663 @@
+| 19    |  900        | 400         |   0         | Stake unlocked; rewards added to AB                   |
+---
+
+
+### **SC Over Epochs**
+
+
+| Epoch | TB   | LB   | AB   | SB   | SC Entries (epoch, stake, locked amount)                                                 | Notes                                               |
+|-------|------|------|------|------|------------------------------------------------------------------------------------------|-----------------------------------------------------|
+| 09    | 1000 |    0 | 1000 |    0 | *Empty*                                                                                  | Initial balance                                     |
+| 10    | 900  |    0 | 500  |  400 | `{10, 400, 0}`, `{11, 400, 400}`, `{12, 400, 400}`, `{13, 400, 400}`, `{14, 400, 400}`   | Staked 400 tokens, withdrew 100 tokens              |
+| 11    | 900  | 400  | 500  |  400 | `{10, 400, 0}`, `{11, 400, 400}`, `{12, 400, 400}`, `{13, 400, 400}`, `{14, 400, 400}`   | Stake locked at start of Entropy Epoch              |
+| 12    | 900  | 400  | 500  |  400 | `{12, 400, 400}`, `{13, 0, 400}`, `{14, 0, 400}`, `{15, 0, 0}`                           | Reduced stake by 400 tokens                         |
+| 13    | 900  | 400  | 500  |  400 | `{12, 400, 400}`, `{13, 0, 400}`, `{14, 0, 400}`, `{15, 0, 0}`                           | Block Production Epoch                              |
+| 14    | 900  | 400  | 500  |    0 | `{12, 400, 400}`, `{13, 0, 400}`, `{14, 0, 400}`, `{15, 0, 0}`                           | Payout Epoch                                        |
+| 15    | 900  | 0    | 400  |  500 | `{15, 500, 0}` , `{16, 500, 500}`, `{17, 500, 500}`, `{18, 500, 500}`, `{19, 500, 500}`  | New Stake of 500                                    |
+| 16    | 900  | 500  | 400  |  500 | `{15, 500, 0}` , `{16, 500, 500}`, `{17, 500, 500}`, `{18, 500, 500}`, `{19, 500, 500}`  |                                                     |
+---
+
+### **Edge Cases and Additional Notes**
+
+- **Adjusting Stake During Staking Epoch:**
+  - Participants can only adjust their stake for the **current staking cycle** during the Staking Epoch.
+  - Once the Staking Epoch ends, the stake commitments are locked till the end of the cycle.
+
+- **Insufficient AB for Adjusting Stake:**
+  - If a participant tries to increase their stake but does not have sufficient AB, they must first increase TB via `deposit()`.
+
+- **Withdrawal Limitations:**
+  - Participants cannot withdraw tokens that are locked in the current staking cycle.
+  - Attempting to withdraw more than AB will result in an error.
+
+- **Automatic Re-Staking:**
+  - Tokens do not automatically unlock, instead if no change in staking is registerd they remain at stake.
+  - Participants must call `adjustStake()` during the next Staking Epoch to free tokens during the next staking cycle.
```

Comment:
This might not make it into the first version.

Reply:
No, probably lots of things not making it into the first version.